File: 55-jump-game/jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        max_reach=0
        s=nums[0]
        for i in range(1,len(nums)):
            if s>0:
                max_reach+=1
                s=s-1
                step=max(nums[i],s)
                s=step
            else:
                return False
        return True            




        

File: 55-jump-game/jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int s=nums[0];
        for(int i=1;i<nums.length;i++){
            if (s>0){
                s=s-1;
                int step=Math.max(s,nums[i]);
                s=step;
            }else{
                return false;
            }
        }
        return true;
    }
}